plasmapy/diagnostics/charged_particle_radiography/detector_stacks.py

```diff
@@ -0,0 +1,286 @@
+        and the
+        `NIST ESTAR database
+        <https://physics.nist.gov/PhysRefData/Star/Text/ESTAR.html>`_.
+
+        Parameters
+        ----------
+        thickness : `~astropy.units.Quantity`
+            The thickness of the layer, in units convertible to meters.
+
+        energy_axis : `~astropy.units.Quantity`
+            The energies corresponding to the stopping power array.
+
+        stopping_power : `~astropy.units.Quantity`
+            The stopping power in the material. Either the linear stopping
+            power (units of J/m) or the mass stopping power
+            (units convertible to J m\ :sup:`^2` / kg) can be provided. If the
+            mass stopping power is provided, the material_density keyword
+            is required.
+
+        mass_density : `~astropy.units.Quantity`, optional
+            The material mass density in units convertible to kg/m\ :sup:`3`.
+            This keyword is required if the provided stopping power is the
+            mass stopping power.
+
+        active : `bool`, default: `True`
+            If `True`, this layer is marked as an active layer.
+
+        name : `str`, optional
+            An optional name for the layer.
+        """
+        self.thickness = thickness
+        self.energy_axis = energy_axis
+        self.active = active
+        self.name = name
+
+        # Handle stopping power provided as either linear or
+        # mass stopping power
+        if stopping_power.unit.is_equivalent(u.J / u.m):
+            self.linear_stopping_power = stopping_power.to(u.J / u.m)
+
+        elif stopping_power.unit.is_equivalent(u.J * u.m**2 / u.kg):
+
+            if mass_density is None:
+                raise ValueError(
+                    "mass_density keyword is required if "
+                    "stopping power is not provided in units "
+                    "convertible to J/m"
+                )
+
+            # Ensure the mass density has the right units
+            try:
+                mass_density = mass_density.to(u.kg / u.m**3)
+            except u.UnitConversionError as e:
+                raise ValueError(
+                    "mass_density keyword must have units convertible to kg/m^3."
+                ) from e
+
+            self.linear_stopping_power = (stopping_power * mass_density).to(u.J / u.m)
+
+        else:
+            raise ValueError(
+                "Units of stopping_power keyword not recognized:"
+                f"{stopping_power.unit}"
+            )
+
+
+class Stack:
+    r"""
+    An ordered list of
+    `~plasmapy.diagnostics.charged_particle_radiography.detector_stacks.Layer` objects.
+
+    Parameters
+    ----------
+    layers : list of `~plasmapy.diagnostics.charged_particle_radiography.detector_stacks.Layer` objects
+        A list of the `~plasmapy.diagnostics.charged_particle_radiography.detector_stacks.Layer`
+        objects that make up the film stack.
```

Comment:
```suggestion
    An ordered list of |Layer| objects.

    Parameters
    ----------
    layers : list of |Layer|
        The objects that make up the film stack.

    .. |Layer| replace:: `~plasmapy.diagnostics.charged_particle_radiography.detector_stacks.Layer`
```
[optional] Since the full namespace is really long, I was wondering if a local reST substition would make the bulk of the docstring more readable (and also fit within ≲72 characters).  This suggestion would also keep type information in the type specification, and the meaning in the parameter description.

Reply:
Yeah good idea, I've now done this for a few other points too